src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -319,6 +327,12 @@ static std::string parseEntitiesVector(ConnectionInfo* ciP, std::vector<EntID>*
       }
       else if (idPatOpt.given)
       {
+        regex_t regex;
+
+        if (regcomp(&regex, idPatOpt.value.c_str(), 0) != 0)
+        {
+          return badInput(ciP, "invalid regexp in subject entities element idPattern");
+        }
         idPattern = idPatOpt.value;
```

Comment:
And here we just might have that same lapsus :-)
If `regex` isn't used after this block, it must be freed.